docs/data/data-grid/ai-assistant/ai-assistant.md

```diff
@@ -0,0 +1,155 @@
+
+{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}
+
+## Processing service integration
+
+Natural language prompts must be processed by a service to understand what kinds of state changes must be applied to the Data Grid to match the user's request.
+You can use MUI's processing service or build your own.
+
+### With MUI's service
+
+The Data Grid provides all the necessary elements for integration with MUI's service.
+
+1. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.
+
+   :::warning
+   Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request on to MUI's service.
+
+   This is an example of a [Fastify proxy](https://www.npmjs.com/package/@fastify/http-proxy) for the prompt requests.
+
+   ```ts
+   fastify.register(proxy, {
+     upstream: 'https://api.mui.com',
+     prefix: '/api/my-custom-path',
+     rewritePrefix: '/api/v1/datagrid/prompt',
+     replyOptions: {
+       rewriteRequestHeaders: (_, headers) => ({
+         ...headers,
+         'x-api-key': process.env.MUI_DATAGRID_API_KEY,
+       }),
+     },
+   });
+   ```
+
+   :::
+
+2. Enable the AI Assistant feature by adding the `aiAssistant` prop.
+   This adds a new button to the Toolbar that opens the Assistant Panel to receive the user's prompts.
+3. Provide the `onPrompt()` callback to pass the user's prompts to the service.
+   The service's response is used internally by the Data Grid to make the necessary state updates.
+
+   :::success
+   You can implement `onPrompt()` with `unstable_gridDefaultPromptResolver()`.
+   This adds the necessary headers and stringifies the body in the correct format for you.
+
+   It also makes it possible to provide additional context for better processing results, as shown below:
+
+   ```ts
+   const PROMPT_RESOLVER_PROXY_BASE_URL =
+     process.env.NODE_ENV === 'development'
+       ? 'http://localhost:3000'
+       : 'https://api.my-proxy.com';
+
+   function processPrompt(query: string, context: string) {
+     const additionalContext = `The rows represent: List of employees with their company, position and start date`;
+
+     return unstable_gridDefaultPromptResolver(
+       `${PROMPT_RESOLVER_PROXY_BASE_URL}/api/my-custom-path`,
+       query,
+       context,
+       additionalContext,
+     );
+   }
+   ```
+
+   :::
+
+4. Provide data examples in either of the following ways:
+   - Fill the `examples` prop in the `columns` array; it's recommended if you want to avoid exposing the row data to the AI assistant.
+   - Provide access to the row data with `allowAiAssistantDataSampling` prop; since actual data is used, it may result in better processing results.
+
+### With a custom service
+
+The Data Grid exposes elements of the AI Assistant feature so you can build your own prompt processing service:
```

Comment:
Do we want to capitalize AI Assistant or not? This doc does both but lowercase seems to be used more. I don't have a strong preference but it should be consistent everywhere. Is this the Proper Noun Name of the feature or is it Ask Your Table?

```suggestion
The Data Grid exposes elements of the AI assistant feature so you can build your own prompt processing service:
```